app.py

```diff
@@ -45,35 +45,73 @@
 # Initialize rate limiting variables
 request_count = 0
 rate_limit_start = time.time()
-
+port_access_token, token_expiry_time, port_headers = None, datetime.now(), {}
 bitbucket_auth = BasicAuth(username=BITBUCKET_USERNAME, password=BITBUCKET_PASSWORD)
-
-# Obtain the access token synchronously
-credentials = {"clientId": PORT_CLIENT_ID, "clientSecret": PORT_CLIENT_SECRET}
-token_response = httpx.post(f"{PORT_API_URL}/auth/access_token", json=credentials)
-port_headers = {"Authorization": f"Bearer {token_response.json()['accessToken']}"}
-
-# Initialize the global AsyncClient with a timeout
 client = httpx.AsyncClient(timeout=httpx.Timeout(60))
 
-async def get_or_create_port_webhook():
-    logger.info("Checking if a Bitbucket webhook is configured on Port...")
+async def get_access_token():
+    credentials = {"clientId": PORT_CLIENT_ID, "clientSecret": PORT_CLIENT_SECRET}
+    token_response = await client.post(f"{PORT_API_URL}/auth/access_token", json=credentials)
+    response_data = token_response.json()
+    access_token = response_data['accessToken']
+    expires_in = response_data['expiresIn']
+    token_expiry_time = datetime.now() + timedelta(seconds=expires_in)
+    return access_token, token_expiry_time
```

Comment:
There could be several reasons where we may get a 401 Unauthorized HTTP error: expired, revoked, malformed, or invalid for other reasons. In this case, handling expiry time can be a bit tricky. I will refer we just return the access token only, and leave the expiry time

Reply:
I kept the time tracking because port continues to return the same JWT token until it expires which will make the refresh redundant